### ISSUE TITLE
Provide script to automate the process

### ISSUE THREAD

**Issue by sarbbottam** [jfmengels/all-contributors-cli#9]
**Please excuse me for cross posting**

Updating contributors w.r.t. .all-contributorsrc is a mechanical activity, I was wondering if it can be automated.

I have submitted a https://github.com/ta2edchimp/opt-cli/pull/20, I guess it could be part of this repo as an utility. If you feel it would be usable please have look at https://github.com/ta2edchimp/opt-cli/pull/20/ and share your thoughts.

Feel free to disagree. Thanks!

/ @ta2edchimp @kentcdodds @jfmengels

**Comment by kentcdodds** [comment]
I think this would be awesome. :+1:

**Comment by jfmengels** [comment]
Well, it kind of already is automated. When you add someone as a contributor or update his contributions using the `add` command, it will also run `generate` afterwards. [Description of the add command here](https://github.com/jfmengels/all-contributors-cli#addupdate-contributors)

For some reason, which I would really like to discuss, the `add` command is not used in your repo (I think you copied the eslint-find-new-rules contributing guidelines which doesn't use it) and IMO, you should use it for the adding a contributor part, as it is a lot of the work into updating your contributor list. 

[The list of steps to contribute](https://github.com/ta2edchimp/opt-cli/blob/master/CONTRIBUTING.md#steps) in your repo is awfully long, discouragingly so. I passed through it once for eslint-find-new-rules, was not pleasant to submit while thinking I may have screwed up any of the steps.

As @jdalton said in the episode of JavaScript Air starring him, you should lower the barrier to contribution for newcomers (which is ironically what opt-cli is all about If I got it right). You could invite the user to use the cli.

Instead of
> Make your changes. Make sure to add yourself to the .all-contributorsrc! More info

say

> Make your changes. Make sure to add yourself to the .all-contributorsrc! Run `npm run add-contributor <your GH username> <contribution types>` More info

(agree that explaining the contribution types is tricky. You could simply tell them to write 'doc'/'code', and update it yourself to the appropriate value).
or don't tell him to do it and do it for him afterwards (if you want his validation/approval, make a PR for the contributor update and make him +1 it). The dream would be [to have a GitHub bot](https://github.com/jfmengels/all-contributors-cli/issues/8#issuecomment-197401269) doing this for you.

Sorry if I totally went off rails on this, as it was not the original intent of the issue.

**tl;dr**: You should add the `add` command, takes away 90% of the mechanical activity, and you should probably reduce the number of steps in your contribution guidelines.
All that said, I am all for having utilities, docs, integration recipes, etc. in this repo if they are useful :+1:

**Comment by ta2edchimp** [comment]
Sad to hear that the contribution guideline seemed so discouraging to you :disappointed: 
How could it be improved? What is it exactly that seemed discouragingly, or is the list just too long?

Would you guys be interested in participating in a more general discussion about this ([I set up a repo and filed an issue to be able to discuss this](https://github.com/ta2edchimp/discussion-contribution-guidelines/issues/1) out of this issue's original discussion).

---

**Regarding the `add` command of `all-contributors-cli`:**

I knew that it exists, but I personally preferred to edit the `.all-contributorsrc` file manually and running `generate` afterwards. This is mainly because I was too lazy to look up the available contribution-type-/emoji-keys, maybe others feel the same?
What about presenting an [`Inquirer.js`](https://github.com/SBoudrias/Inquirer.js/) based prompt when running `all-contributors-cli` without any arguments or when specifying `add` but lacking a username and/or contribution type?

**Comment by jfmengels** [comment]
> How could it be improved? What is it exactly that seemed discouragingly, or is the list just too long?

The list is just too long. I think beginners may get scared to miss a step and be yelled at. Doesn't matter how much it gets decorated by gifs, emojis or encouraging talk, a huge list looks like a wall that is hard to cross over.
It is especially long and useless (some parts of it I mean) if you want to make a small change like fix a typo. I think some steps could be removed, even if that lowers the quality of the PR, and make the maintainers fix the easy mistakes (similar to what's being done on Lodash). Obviously that takes more time for maintainers, but it's a potential trade-off.

> Would you guys be interested in participating in a more general discussion?

Sure! Will check it out later when I have more time :)

> What about presenting an Inquirer.js based prompt when running all-contributors-cli without any arguments?

Yeah! Great idea! :heart:

**Comment by jfmengels** [comment]
Also, I guess I didn't take a deep enough look at the proposed utility (the part where the generated gets commited and pushed), my bad. That makes it obviously more interesting. I'll improve the `add` command, and you guys let me know if this still makes to you, and we'll see about adding it in a recipe or something.

**Comment by ta2edchimp** [comment]
>>Would you guys be interested in participating in a more general discussion?

>Sure! Will check it out later when I have more time :)

That would be awesome! Take your time, there's a wall of text over there. But I'd really like to address this sooner or later, as I'd like to lower the barrier (esp. for such a small projects, like `opt-cli`) as much as possible, while still ensuring everything to be as cleaned up and orderly as possible (but in the end, that could just be a typical german attitude :stuck_out_tongue_winking_eye:).

And, sorry for hijacking the initial issue and letting it derail.

**Comment by jfmengels** [comment]
> And, sorry for hijacking the initial issue and letting it derail.

This idea makes it all worth it :heart: :
> What about presenting an Inquirer.js based prompt when running all-contributors-cli without any arguments?

**Comment by sarbbottam** [comment]
The intention behind `script to automate` was to get rid of the manual commit.

I would be great to achieve it by any means, as suggested by @ta2edchimp, use [`Inquirer.js`](https://github.com/SBoudrias/Inquirer.js/) and auto `commit` once the README is updated

**Comment by jfmengels** [comment]
I left a comment [here](https://github.com/jfmengels/all-contributors-cli/issues/10#issuecomment-203602456)

@sarbbottam If this suffices you, do you mind closing this issue? If you want something more, please let me know ;)

**Comment by jfmengels** [comment]
Well that's a good sign :)
